crates/stages/src/error.rs

```diff
@@ -76,6 +76,36 @@ pub enum StageError {
     /// rely on external downloaders
     #[error("invalid download response: {0}")]
     Download(#[from] DownloadError),
+    /// Database is ahead of snapshot data.
+    #[error("missing snapshot data for block number: {number}", number = block.number)]
+    MissingSnapshotData {
```

Comment:
can we make these a new error variant specifically for snapshots/static files, but perhaps this makes error handling worse, so this is just a question, seems fine